pilot/pkg/bootstrap/options.go

```diff
@@ -17,78 +17,59 @@ package bootstrap
 import (
 	"time"
 
-	meshconfig "istio.io/api/mesh/v1alpha1"
 	"istio.io/pkg/ctrlz"
 	"istio.io/pkg/env"
 
 	"istio.io/istio/pilot/pkg/features"
 	kubecontroller "istio.io/istio/pilot/pkg/serviceregistry/kube/controller"
 	"istio.io/istio/pkg/config/constants"
-	istiokeepalive "istio.io/istio/pkg/keepalive"
+	"istio.io/istio/pkg/keepalive"
 )
 
-// MeshArgs provide configuration options for the mesh. If ConfigFile is provided, an attempt will be made to
-// load the mesh from the file. Otherwise, a default mesh will be used with optional overrides.
-type MeshArgs struct {
-	ConfigFile string
-	// Used for test
-	MixerAddress string
-}
-
-// ConfigArgs provide configuration options for the configuration controller. If FileDir is set, that directory will
+// RegistryOptions provide configuration options for the configuration controller. If FileDir is set, that directory will
 // be monitored for CRD yaml files and will update the controller as those files change (This is used for testing
 // purposes). Otherwise, a CRD client is created based on the configuration.
-type ConfigArgs struct {
-	ControllerOptions          kubecontroller.Options
+type RegistryOptions struct {
+	// If FileDir is set, the below kubernetes options are ignored
+	FileDir string
+
+	Registries []string
+
+	// Kubernetes controller options
+	KubeOptions kubecontroller.Options
+	// ClusterRegistriesNamespace specifies where the multi-cluster secret resides
 	ClusterRegistriesNamespace string
 	KubeConfig                 string
-	FileDir                    string
+
+	// Consul options
+	ConsulServerAddr string
 
 	// DistributionTracking control
 	DistributionCacheRetention time.Duration
 
-	DisableInstallCRDs bool
-
 	// DistributionTracking control
 	DistributionTrackingEnabled bool
 }
 
-// ConsulArgs provides configuration for the Consul service registry.
-type ConsulArgs struct {
-	ServerURL string
-}
-
-// ServiceArgs provides the composite configuration for all service registries in the system.
-type ServiceArgs struct {
-	Registries []string
-	Consul     ConsulArgs
-}
-
 // PilotArgs provides all of the configuration parameters for the Pilot discovery service.
 type PilotArgs struct {
```

Comment:
Should we call this IstiodArgs instead of PilotArgs now? I have not looked at the rest of the PR but since you are any way making this change, thought we can change this as well. WDYT?

Reply:
I am not sure it is worth doing that since pilot is well known, even the binary is called `pilot-discovery`

Reply:
We are trying to move away from 'pilot' - the service is now called istiod. The binary can be renamed as well.

But probably not worth for this PR.